src/shared_utilities/Domains.jl

```diff
@@ -72,24 +74,55 @@ end
 """
     Column(;
         zlim::Tuple{FT, FT},
-        nelements::Int) where {FT}
-Outer constructor for the `Column` type.
+        nelements::Int,
+        dz_tuple::Union{Tuple{FT, FT}, Nothing} = nothing) where {FT}
+
+Outer constructor for the `Column` type. 
+
+Using `ClimaCore` tools, the coordinate
+mesh can be stretched such that the top of the domain has finer resolution
+than the bottom of the domain. In order to activate this, a tuple with the
+target dz_bottom and dz_top should be passed via keyword argument. The default is
+uniform spacing. Please note that in order to use this feature, ClimaCore requires that
+the elements of zlim be <=0. Additionally, the dz_tuple you supply may not be compatible
+with the domain boundaries in some cases, in which case you may need to choose
+different values.
+
 The `boundary_tags` field values are used to label the boundary faces 
 at the top and bottom of the domain.
 """
-function Column(; zlim::Tuple{FT, FT}, nelements::Int) where {FT}
+function Column(;
+    zlim::Tuple{FT, FT},
+    nelements::Int,
+    dz_tuple::Union{Tuple{FT, FT}, Nothing} = nothing,
+) where {FT}
     @assert zlim[1] < zlim[2]
     boundary_tags = (:bottom, :top)
     column = ClimaCore.Domains.IntervalDomain(
         ClimaCore.Geometry.ZPoint{FT}(zlim[1]),
         ClimaCore.Geometry.ZPoint{FT}(zlim[2]);
         boundary_tags = boundary_tags,
     )
-    mesh = ClimaCore.Meshes.IntervalMesh(column; nelems = nelements)
```

Comment:
I was wondering if you created a new method for `Column`, but I see you have a `if` and no new method. 
I have a simple question / something to consider: would a new method be computationally more efficient? (faster run), or is this function just called once and basically this changes nothing?

Reply:
Good question! I wasnt sure, I tried making a minimum working example and in this simple case the execution time was very similar.
```
module Tmp

struct FlagTrue end
struct FlagFalse end

f(x::FlagTrue) = 1
f(x::FlagFalse) = 2
function g(x)
    if typeof(x) == FlagTrue
        return 1
    else
        return 2
    end
end


end

using .Tmp
using BenchmarkTools
@btime Tmp.f(Tmp.FlagTrue())
@btime Tmp.g(Tmp.FlagTrue())
```

I asked @dennisYatunin and he said that in more recent versions of Julia using an if statement should be the same as writing a new method. You also made a good point about this being only executed once, so it might be okay even if it was slower to use the if statement.